### ISSUE TITLE
Object creation is erroneously explained with a copy construction

### ISSUE THREAD

**Issue by GPopov** [andreasfertig/cppinsights#120]
Creating an object of some class is explained as if a copy construction is performed, which is not the case for standard-compliant compilers.

Example:

```
struct A
{
    A() = default;
    A(const A&) = delete;
};

int main()
{
    A a; //1
}
```

The line (1) is transformed into:
`A a = A();`

This is wrong, as it would invoke the deleted copy ctor of A. Proof: https://godbolt.org/z/wct9r5

**Comment by andreasfertig** [comment]
Hello @GPopov,

thank you for reporting that finding. How would you like the default ctor to be visualized?

Andreas

**Comment by GPopov** [comment]
This is an interesting and difficult question. If there was a obviously really more expressive way, most probably the C++ committee would have opted for it already. A very suboptimal soultion would be as a comment, detailing all the relevant c-tors and special member functions that get invoked. 
Something along the lines of 
```
A a;
A a2 = a;
A a3 = A();
```

transformed into
```
//A::A()
A a;

//A::A(const A&)
A a2 = a;
// 1. A::A() 2. A::(const A&) 
A a3 = A();
```

This solution has multiple problems with it, starting with the fact that if the line is rather more complicated, this comment would quickly just as difficult to understand as the original line. Maybe some plain-english explanation could be thought about, again as a comment, but this feels a bit against the goal of your project. 

As a starting point, I'd say that at the very least and a starting point, no incorrect explanation should be produced.

**Comment by andreasfertig** [comment]
Yes it is indeed difficult. I agree that nothing incorrect should be produced. On the other hand, without a proper alternative I feel reluctant to just remove it. I like to think more about it.
